### PR TITLE
Adds slides projectors

### PR DESCRIPTION
**Pull request by comma**
Handheld devices that can accept small or tiny items.
Then you can click anywhere to project image of the active item to that location.
Projections can be examined, and will forward examine call to object with distance of 1, so you can see photos and read papers.
Clicking projector in hand opens the UI that lets you select active slide.
https://a.uguu.se/1Ia24OVvj6wx_2020-04-18_03-45-52.mp4

### CODE REVIEW

File: code/game/machinery/slide_projector.dm [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	current_slide = null
+	stop_projecting()
+	. = ..()
+
+/obj/item/storage/slide_projector/on_update_icon()
+	icon_state = "projector[!!projection]"
+
+/obj/item/storage/slide_projector/get_mechanics_info()
+	. = ..()
+	. += "Click in hand to open the interface."
```
Comment (MistakeNot4892) [inline code/game/machinery/slide_projector.dm]: 'Use in hand' maybe?

File: code/game/machinery/slide_projector.dm [hunk 2]
```diff
@@ -0,0 +1,189 @@
+/obj/item/storage/slide_projector
```
Comment (MistakeNot4892) [inline code/game/machinery/slide_projector.dm:1]: Do you need that flag set so you don't beat things with it if you use it from adjacent?

File: code/game/machinery/slide_projector.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	for(var/T in projection_types)
+		if(istype(current_slide, T))
+			projection_type = projection_types[T]
+			break
+	projection = new projection_type(target)
+	projection.set_source(current_slide)
+	GLOB.moved_event.register(src, src, .proc/check_projections)
+	update_icon()
+
+/obj/item/storage/slide_projector/attack_self(mob/user)
```
Comment (MistakeNot4892) [inline code/game/machinery/slide_projector.dm:76]: Minor thought, if you ever made one of those projectors Deer linked in the chat, you could check anchored on attack_hand() to redirect to interact() as well.

File: code/game/machinery/slide_projector.dm [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	MA.layer = FLOAT_LAYER
+	MA.appearance_flags = RESET_ALPHA
+	MA.alpha = 170
+	MA.pixel_x = 0
+	MA.pixel_y = 0
+	overlays |= MA
+
+/obj/effect/projection/proc/set_source(obj/item/I)
+	source = weakref(I)
+	desc = "It's currently showing \the [I]"
```
Comment (MistakeNot4892) [inline code/game/machinery/slide_projector.dm]: Full stop needed.

**Review by MistakeNot4892 — COMMENTED** [review]
Other than tiny grammar shit, looks good.

File: code/game/machinery/slide_projector.dm (hunk 2 above)
Comment (comma) [reply code/game/machinery/slide_projector.dm:1]: Actually works like that already, clicked on adjacent computer and it just projected there.

File: code/game/machinery/slide_projector.dm (hunk 3 above)
Comment (comma) [reply code/game/machinery/slide_projector.dm:76]: I just figured that this is already useless enough feature and making it tied ot specific location makes chance no one would ever use it to 100% cue handheld

File: code/game/machinery/slide_projector.dm (hunk 4 above)
Comment (comma) [reply code/game/machinery/slide_projector.dm]: Fixed

File: code/game/machinery/slide_projector.dm (hunk 1 above)
Comment (comma) [reply code/game/machinery/slide_projector.dm]: Fixed

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**